pretyping/reductionops.ml

```diff
@@ -1192,11 +1192,15 @@ let vm_infer_conv ?(pb=Reduction.CUMUL) env t1 t2 =
 
 let default_plain_instance_ident = Id.of_string "H"
 
+type subst_fun = { sfun : metavariable -> EConstr.t }
```

Comment:
Is this record necessary? Please add a comment if so.

Reply:
It's not strictly necessary, but I find it cleaner to wrap this function in a record.

Reply:
can you explain why?

Reply:
I guess it's just a matter of taste, I usually find it fairly ugly to pass anonymous higher order functions to deep calls such as `plain_instance`. Wrapping this in a record makes explicit that this function is supposed to be the meta expander, and it makes it easier in the future to change the type (e.g. if we want to take the lift argument) without debugging type errors at the wrong place. I have PTSD from some parts of the code that gladly use crazy HO calls in spaghetti code, trying to tweak this kind of code is a nightmare.

Reply:
Also here we are manipulating a mutable state, so putting barriers between closures ensure statically that nobody will ever move the creation of the reference at the wrong place w.r.t. the closure.

Reply:
I won't argue furhter